utils/conda-utils.js

```diff
@@ -0,0 +1,88 @@
+/*eslint-env node*/
+/*eslint-disable no-console*/
+'use strict';
+const {spawnSync, spawn} = require('child_process'),
+    os = require('os'),
+    path = require('path'),
+    fs = require('fs'),
+    yaml = require('js-yaml'),
+    CONDA_COMMAND = 'conda';
+
+const getCondaEnvs = function () {
+    const envProcess = spawnSyncCondaProcess(['env', 'list']);
+    return envProcess.stdout.toString().split('\n')
+        .filter(line => !!line && !line.startsWith('#'))
+        .map((env) => {
+            const [name, path] = env.split(/\s+/);  //eslint-disable-line no-unused-vars
+            return name;
+        }).filter(env => !!env);
+};
+
+const envExists = function (name) {
+    const availableEnvs = getCondaEnvs();
+    return availableEnvs.indexOf(name) > -1;
+};
+
+const dumpYAML = function (environment, envFileName) {
+    if (!envFileName) {
+        envFileName = path.join(os.tmpdir(), 'deepforge.yml');
+    }
+    const envYamlString = yaml.safeDump(environment);
+    fs.writeFileSync(envFileName, envYamlString, 'utf8');
+    return envFileName;
+};
+
+const checkConda = function () {
+    const conda = spawnSyncCondaProcess(['-V']);
+    if (conda.status !== 0) {
+        throw new Error(`Please install conda before continuing. ${conda.stderr.toString()}`);
+    }
+};
+
+
+const createOrUpdateEnvironment = function (envFile) {
+    const env = yaml.safeLoad(fs.readFileSync(envFile, 'utf8'));
+    if (process.env.DEEPFORGE_CONDA_ENV && process.env.DEEPFORGE_CONDA_ENV !== env.name) {
```

Comment:
This file seems like it is written to be used as a module rather than just a cli script. If so, this probably shouldn't rely on an environment variable in an internal method as it can have some unintended consequences when using it programmatically.